cfbs/pretty_json.py

```diff
@@ -0,0 +1,114 @@
+import json
+from typing import OrderedDict
+
+
+def pretty(s):
+    MAX_LEN = 80
+    INDENT_SIZE = 2
+
+    def _force_wrap(parent):
+        if len(parent) < 2:
+            return False
+        for child in parent:
+            if not isinstance(child, (tuple, list, dict)):
+                return False
+            if len(child) < 2:
+                return False
+        return True
+
+    def _encode_list(lst, indent, cursor):
+        if not lst:
+            return "[]"
+
+        if not _force_wrap(lst):
+            buf = "["
+            first = True
+            for value in lst:
+                if len(buf) > MAX_LEN:
+                    break
+                if first:
+                    first = False
+                else:
+                    buf += ", "
+                buf += _encode(value, indent, cursor + len(buf))
+            buf += "]"
+
+            if indent + cursor + len(buf) <= MAX_LEN:
+                return buf
```

Comment:
Unclear whether buf can be multiple lines or not. Please change to json.dumps, or add asserts and map/join.